addon/helpers/optional.js

```diff
@@ -1,8 +1,8 @@
 import { helper } from 'ember-helper';
 
-export function optional([action]) {
+export function optional([action, ...args]) {
   if (typeof action === 'function') {
-    return action;
+    return action.bind(this, ...args);
```

Comment:
Yeah; I'm not sure what it should be, either. Honestly, I'm having a hard time even finding an example of a closure action with a custom context.

Regardless of the test case, I think this could be solved by returning a closure of the action and doing away with any `bind()` silliness altogether; would you agree?

```js
if (typeof action === 'function') {
  return () => action(...args);
}
```